src/main.py

```diff
@@ -1262,11 +1254,17 @@ def _return_to_cluster(self, new_primary, role, is_dead=False):
             # rewinding and failed. So only hard way possible in this case.
             #
             last_op = self.zk.noexcept_get('%s/%s/op' % (self.zk.MEMBERS_PATH, helpers.get_hostname()))
-            logging.info('Last op is: %s' % str(last_op))
-            if role != 'primary' and not self.is_op_destructive(last_op) and not self._is_simple_primary_switch_tried():
-                logging.info('Trying to do a simple primary switch.')
+            if role == 'primary' or self.is_op_destructive(last_op) or self._is_simple_primary_switch_tried():
+                logging.info('Could not do a simple primary switch')           
+                logging.debug('Possible reasons: Role: %s, Last op is destructive: %s, Simple primary switch tried: %s',
+                    role, self.is_op_destructive(last_op), self._is_simple_primary_switch_tried()
```

Comment:
`self._is_simple_primary_switch_tried()` call is fetching data from ZK.
Calling it for logging only:
- slows down everything
- gives unpredictably different result from one that was used in `if` statement